### ISSUE TITLE
Evaluation result is incorrect

### ISSUE THREAD

**Issue by hhqweasd** [wengzejia1/Open-VCLIP#12]
Hi, I trained Open-VCLIP following the readme, and the end of the training log is as follows:

 05:39:51][INFO] logging.py:   99: json_stats: {"_type": "val_iter", "epoch": "22/22", "eta": "0:01:18", "gpu_mem": "25.75G", "iter": "100/155", "time_diff": 1.41916541, "top1_err": 21.09375000, "top5_err": 6.25000000}
[01/18 05:40:00][INFO] logging.py:   99: json_stats: {"_type": "val_iter", "epoch": "22/22", "eta": "0:00:18", "gpu_mem": "25.75G", "iter": "110/155", "time_diff": 0.40004537, "top1_err": 20.31250000, "top5_err": 4.68750000}
[01/18 05:40:07][INFO] logging.py:   99: json_stats: {"_type": "val_iter", "epoch": "22/22", "eta": "0:00:17", "gpu_mem": "25.75G", "iter": "120/155", "time_diff": 0.49999434, "top1_err": 21.09375000, "top5_err": 7.03125000}
[01/18 05:40:16][INFO] logging.py:   99: json_stats: {"_type": "val_iter", "epoch": "22/22", "eta": "0:00:12", "gpu_mem": "25.75G", "iter": "130/155", "time_diff": 0.49982308, "top1_err": 22.65625000, "top5_err": 8.59375000}
[01/18 05:40:25][INFO] logging.py:   99: json_stats: {"_type": "val_iter", "epoch": "22/22", "eta": "0:00:49", "gpu_mem": "25.75G", "iter": "140/155", "time_diff": 3.29047562, "top1_err": 21.87500000, "top5_err": 6.25000000}
[01/18 05:40:31][INFO] logging.py:   99: json_stats: {"_type": "val_iter", "epoch": "22/22", "eta": "0:00:01", "gpu_mem": "25.75G", "iter": "150/155", "time_diff": 0.34571415, "top1_err": 19.53125000, "top5_err": 6.25000000}
[01/18 05:40:33][INFO] logging.py:   99: json_stats: {"RAM": "89.90/754.06G", "_type": "val_epoch", "epoch": "22/22", "gpu_mem": "25.75G", "min_top1_err": 20.21018592, "min_top5_err": 7.22514146, "time_diff": 0.90270419, "top1_err": 20.96806790, "top5_err": 7.23524656}
[01/18 05:40:33][INFO] train_net.py: 1420: training done: _p149.62_f152.11 _t32.45_m25.75 _a79.79 Top5 Acc: 92.77 MEM: 25.75 f: 152.1058

After training, I got the checkpoint files from epoch_00001 to epoch_00022.pyth. Then I ran weight_average_tool.py and obtained swa_2_22.pth. 
However, when I used this checkpoint to evaluate, the top-1 accuracy and top-5 accuracy are extremely low, I got:
1/25 14:18:32][INFO] logging.py:   99: json_stats: {"cur_iter": "164", "eta": "0:00:04", "split": "test_iter", "time_diff": 1.16411489}
[01/25 14:18:33][INFO] logging.py:   99: json_stats: {"cur_iter": "165", "eta": "0:00:03", "split": "test_iter", "time_diff": 1.14636344}
[01/25 14:18:35][INFO] logging.py:   99: json_stats: {"cur_iter": "166", "eta": "0:00:02", "split": "test_iter", "time_diff": 1.14615193}
[01/25 14:18:35][INFO] logging.py:   99: json_stats: {"cur_iter": "167", "eta": "0:00:00", "split": "test_iter", "time_diff": 0.60836913}
[01/25 14:18:36][INFO] test_net.py:  196: Successfully saved prediction results to /root/flash/Open-VCLIP-main/script/training/temporalclip_vitb16_8x16_interpolation_bugfix_0.5ratio_rand0.0_0.6sample3/temp.pyth
[01/25 14:18:36][INFO] logging.py:   99: json_stats: {"split": "test_final", "top1_acc": "3.93", "top5_acc": "13.13"}
[01/25 14:18:36][INFO] test_net.py:  448: Finalized testing with 3 temporal clips and 1 spatial crops
[01/25 14:18:36][INFO] test_net.py:  470: _p149.62_f152.10_3a3.93 Top5 Acc: 13.13 MEM: 15.88 f: 152.1046
[01/25 14:18:36][INFO] test_net.py:  471: _p149.62_f152.10_3a3.93

The log shows that the top_1 ACC is only 3.93.
When I change the generated checkpoint  swa_2_22.pth to the pre-trained checkpoint download from your google drive, the evaluation results are normal. 
My question is, is there a problem with my training, or is there something wrong with saving the weight average file?

**Comment by wengzejia1** [comment]
It is strange and I have never met such phenomenon.

Maybe you can try to debug by running the ''weight_average_tool.py''  with the setting of **''wa_start = 22''** and **''wa_end = 22'**' to get a swa_22_22.pth, and then test the performance on the ''swa_22_22.pth'' and ''epoch_00022.pyth'', to see whether the performance is the same.

**Comment by hhqweasd** [comment]
I tried swa_22_22.pth but the results on ucf-101 dataset and the top-1 ACC is about 3.73%.
I evaluated the swa_2_22.pth&swa_22_22.pth on the K400 dataset, and the results are fine and the top-1 ACC is 80%+.
I also evaluated the swa_2_22.pth on the K600 val, and the top-1 ACC is about 3.23%.
The training dataset is the compressed k400 dataset. 
It seems that the evaluation results are normal if and only if both the training and the testing set are the K400 dataset.

**Comment by wengzejia1** [comment]
what about setting "wa_start" as 10, 20, 21 respectively.

**Comment by wengzejia1** [comment]
`Your Reply: ''I also evaluated the swa_2_22.pth on the K600 val, and the top-1 ACC is about 3.23%.'' `
Do you mean when you use your own swa_2_22.pth checkpoint, the UCF or K600 performance is low, but when you use the google drive provided ''swa_2_22.pth'' the UCF or K600 performance is normal ?

**Comment by hhqweasd** [comment]
> `Your Reply: ''I also evaluated the swa_2_22.pth on the K600 val, and the top-1 ACC is about 3.23%.'' ` Do you mean when you use your own swa_2_22.pth checkpoint, the UCF or K600 performance is low, but when you use the google drive provided ''swa_2_22.pth'' the UCF or K600 performance is normal ?

Yes

**Comment by wengzejia1** [comment]
That's wired. How about the score when you test the ''epoch_00022.pyth'' directly ? The score shouldn't be so low.

By the way, from your log file above, it seems you are running the CLIP L/14 based model ? If so, when you run the "weight_average_tool.py", have you changed the parameter "raw_clip = '/root/.cache/clip/ViT-B-16.pt' " to "raw_clip = '/root/.cache/clip/ViT-L/14.pt' "?

**Comment by hhqweasd** [comment]
The top1 acc of "epoch_00022.pyth" on k600 is 3.02%. The number of parameters of the model is 149.62M, which is lower than the parameters of ViT-L. I checked all the config files where the raw_clip is ViT-B-16.
Now it seems that it is a training issue, could it be overfitting on k400 and causing it to not work on the other datasets？

**Comment by wengzejia1** [comment]
Hi, I rerun my code and the scores are normal. 

I have uploaded the original log file "stdout (1).log" in the [google drive](https://drive.google.com/file/d/1FPGZh8PREyarv2M6-u2bx6TA-LpP75Iy/view?usp=drive_link), you can compare that with yours.

And maybe you can firstly try to set the "TRAIN.LINEAR_CONNECT_CLIMB" in training script as False to check whether scores on K600/UCF/HMDB are still so low. You to set the "TEST.PATCHING_MODEL" as False in the testing script and test the checkpoints along the training process (checkpoint 0, 1, ... to 22) to see how the scores change. The table 4 in the Open-VCLIP paper shows the overfitting problem, that final scores might be lower than the original CLIP on some dataset if directly fine-tuning CLIP, but will not be so low as you test. 

By the way, do you have WeChat? You can add me on WeChat to communicate more immediately. My WeChat ID: wengzejia971002.

**Comment by hhqweasd** [comment]
Thank you for sharing the log file! Some of my configs are not the same as yours. I'll fix my config error and train again to see if there is still such an issue.
![diff](https://github.com/wengzejia1/Open-VCLIP/assets/46904457/455ccc29-ce2e-4b76-bafa-987cf9e3bebc)
